### ISSUE TITLE
Named imports not used when clashing with non-object globals

### ISSUE THREAD

**Issue by rossipedia** [web-infra-dev/rspack#5166]
### System Info

```
  System:
    OS: macOS 14.0
    CPU: (8) arm64 Apple M2
    Memory: 960.00 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/Library/Caches/fnm_multishells/85940_1704050315190/bin/node
    Yarn: 1.22.21 - ~/Library/Caches/fnm_multishells/85940_1704050315190/bin/yarn
    npm: 10.2.5 - ~/Library/Caches/fnm_multishells/85940_1704050315190/bin/npm
    pnpm: 8.13.1 - ~/Library/Caches/fnm_multishells/85940_1704050315190/bin/pnpm
  Browsers:
    Safari: 17.0
  npmPackages:
    @rspack/cli: ^0.4.5 => 0.4.5 
    @rspack/core: ^0.4.5 => 0.4.5 
```

### Details

When using a named import that is one of the [value-typed global properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects#value_properties) (eg: `Infinity`, `NaN`, etc), it seems as though rspack ignores the name assigned to the import when referencing it and instead embeds a reference to the global property. It also seems to do this inconsistently.

Given the following source:

```js
import { Array, Infinity, NaN, Object, globalThis, undefined } from './mod';

export function render() {
  Array();
  Infinity();
  NaN();
  Object();
  globalThis();
  undefined();
 
  ...
}
```

rspack produces the following output:

```js
 function render() {
    (0, _mod__WEBPACK_IMPORTED_MODULE_1__.Array)();
    Infinity();
    NaN();
    (0, _mod__WEBPACK_IMPORTED_MODULE_1__.Object)();
    (0, _mod__WEBPACK_IMPORTED_MODULE_1__.globalThis)();
    undefined();
    
    ...
}
```

Working in webpack:

<img width="752" alt="image" src="https://github.com/web-infra-dev/rspack/assets/96213/192ba083-8eff-48d9-9ade-3993f667f85d">

Broken in rspack:

<img width="752" alt="image" src="https://github.com/web-infra-dev/rspack/assets/96213/2f3a078b-4375-42be-8f95-83be236e49fa">

### Reproduce link

https://github.com/rossipedia/rspack-infinity-export-bug

### Reproduce Steps

1. Run `npm run dev:webpack`
2. Open http://localhost:8080
3. See the various console.log outputs in browser console

4. Run `npm run dev:rspack`
5. Open http://localhost:80800
6. See a console error: `TypeError: Infinity is not a function`

**Comment by hardfist** [comment]
tracked in https://github.com/swc-project/swc/issues/8465

**Comment by xc2** [comment]
fix has been included in swc_ecma_transforms_base 0.135.11 and rspack 0.5.2.

please try out the latest version of rspack.
